### DISCUSSION TITLE
Remix - Adonisjs

### DISCUSSION THREAD

**Discussion by yarapolana** [adonisjs/core#3985] [General]
Given the rise of these newish frameworks, has anyone tried to connect Remix Run with AdonisJS?

**Comment by Barbapapazes** [comment]
Hello, why doing this?

**Reply by yarapolana** [reply]
Remix is a great framework that has ability to create full stack applications just like Adonis, they are more inclined to front end and Adonis to backend. I would like to mix them together if possible.
I already have separate Adonis and Remix projects and was wondering if anyone has managed to use them both as one.

**Reply by Barbapapazes** [reply]
And what's your problematic? Because with tools like Unpoly or HTMX or Inertia, you can build a product that use the best of the backend and the front-end world!

**Reply by yarapolana** [reply]
Maybe you misunderstood me. I am just opening a conversation to see if anyone has done it and continue from there not to get help in implementing it.😊

**Comment by ditorodev** [comment]
I have never worked with adonis, but i get your point and im also interested on something like this. Recently been using the Epic Stack because i dont wanna focus on the same authentication and caching and even models problems, if adonis can handle most of the backend so it is just querying and returning stuff I wanna try out with remix

**Comment by dimitrisnl** [comment]
There's nothing odd about that combo. The usual suspect is sharing types between the two if you're using TS. Other than that it's smooth sailing.

**Comment by GabrielNBDS** [comment]
they are two servers, running them together would be the same as running remix and next together. since they're not just libs that you can add to any project I'd just run them separately, making both run in one server would be too much effort

**Comment by dimitrisnl** [comment]
For anyone revisiting the thread

1. Run them separately
2. Use tokens to communicate server to server

**Comment by Julien-R44** [comment]
just wanted to add a quick thing : i'm not at all familiar with Remix. i have never tried it, but a while ago I came across this page: 

- https://remix.run/docs/en/main/start/quickstart#bring-your-own-server
- https://remix.run/docs/en/main/discussion/runtimes#adapters

basically: you can plug your own server into Remix. so I think it should be possible and relatively easy to run Remix and Adonis in a single Node process. all you have to do is make an adapter, like the one made for express, which should be fairly simple to do. and that would allow you to use Adonis and its features directly in your Remix "loaders": https://remix.run/docs/en/main/route/loader

i tried the same thing recently but with [Vike](https://vike.dev/) instead of Remix, which has the same principle

basically, you can also plug in your own server, be it Express, Fastify or Adonis. worked great : it allowed me to use Adonis directly in my Vike API routes / Data fetching ( this is the Remix loaders equivalent )

i will give it a try when i've got the time, if by then nobody's done it

**Comment by jarle** [comment]
I built this some time ago for my SaaS when I realized that it was actually possible to embed Remix within AdonisJS.

Julien and the core team has done a great job at integrating AdonisJS with Vite (Remix also uses Vite), so now the combination works even better than when I first built it 🙌

I have made it open source and available here:

- https://github.com/jarle/remix-adonisjs#readme

It's still a bit experimental, so feel free to check it out and report any bugs/issues you find.

**Reply by yarapolana** [reply]
This is awesome @jarle Thank you for sharing.
My two fav frameworks together.
I will close this for now.

**Comment by yarapolana** [comment]
Thank you for the wonderful conversations I will close this for now.

**Reply by RomainLanz** [reply]
I would recommend that you also have a look at our [InertiaJS integration](https://adonisjs.com/blog/experimental-release-for-inertia). 👍🏻
